corehq/pillows/domain.py

```diff
@@ -59,6 +59,9 @@ class DomainReindexerFactory(ReindexerFactory):
     ]
 
     def build(self):
+        args = ElasticPillowReindexer.__init__.__code__.co_varnames
+        # Drop options that are not kwargs of the reindexer (like "reset")
+        options = {k: v for k, v in self.options.items() if k in args}
```

Comment:
This filters out invalid keyword arguments before trying to pass them to `ElasticPillowReindexer.__init__()`.

Another option would be to use  `options.pop('reset')`. I chose against that because it felt too specific and hacky. But conversely this approach feels too generic.

Alternatively, all reindexers' `__init__()` methods could take **kwargs, and ignore them. ... Hmmm ... maybe that's best.

Reply:
> all reindexers' `__init__()` methods could take **kwargs, and ignore them.

I'm not a fan of that approach because it's an easy source of bugs: someone passes in an argument they think is important when it's actually being ignored, which leads to unexpected program behavior. My preference is to pop the ones you expect but explicitly want to ignore, and raise an error for any other unexpected arguments.

As [the saying](https://www.python.org/dev/peps/pep-0020/) goes: _Errors should never pass silently._ (_Unless explicitly silenced._)